python/ray/train/examples/horovod/horovod_cifar_pbt_example.py

```diff
@@ -111,14 +112,17 @@ def train_loop_per_worker(config):
             if config["smoke_test"]:
                 break
 
-        checkpoint = Checkpoint.from_dict(
-            dict(
+        from tempfile import TemporaryDirectory
+
+        with TemporaryDirectory() as tmpdir:
+            state_dict = dict(
                 model_state=net.state_dict(),
                 optimizer_state=optimizer.state_dict(),
                 epoch=epoch,
             )
-        )
-        train.report(dict(loss=running_loss / epoch_steps), checkpoint=checkpoint)
+            torch.save(state_dict, f"{tmpdir}/checkpoint.bin")
```

Comment:
can we only use `os.path.join` for all examples?

Reply:
Hopefully that's easy to do with find and replace 😅

Reply:
ahh ok. Let me replace them,